code/modules/mining/equipment_locker.dm

```diff
@@ -484,28 +484,58 @@
 	throw_speed = 3
 	throw_range = 5
 	origin_tech = "bluespace=2"
+	slot_flags = SLOT_BELT
 
 /obj/item/device/wormhole_jaunter/attack_self(mob/user)
+	user.visible_message("<span class='notice'>[user.name] activates the [src.name]!</span>")
+	activate(user)
+
+/obj/item/device/wormhole_jaunter/proc/turf_check(mob/user)
 	var/turf/device_turf = get_turf(user)
 	if(!device_turf||device_turf.z==2||device_turf.z>=7)
 		user << "<span class='notice'>You're having difficulties getting the [src.name] to work.</span>"
+		return FALSE
+	return TRUE
+
+/obj/item/device/wormhole_jaunter/proc/activate(mob/user)
+	if(!turf_check(user))
 		return
+
+	var/list/L = list()
+	for(var/obj/item/device/radio/beacon/B in world)
+		var/turf/T = get_turf(B)
+		if(T.z == ZLEVEL_STATION)
+			L += B
+	if(!L.len)
+		user << "<span class='notice'>The [src.name] found no beacons in the world to anchor a wormhole to.</span>"
+		return
+	var/chosen_beacon = pick(L)
+	var/obj/effect/portal/wormhole/jaunt_tunnel/J = new /obj/effect/portal/wormhole/jaunt_tunnel(get_turf(src), chosen_beacon, lifespan=100)
+	J.target = chosen_beacon
+	try_move_adjacent(J)
+	playsound(src,'sound/effects/sparks4.ogg',50,1)
+	qdel(src)
+
+/obj/item/device/wormhole_jaunter/emp_act(power)
+	var/triggered = FALSE
+
+	if(usr.get_item_by_slot(slot_belt) == src)
+		if(power == 1)
+			triggered = TRUE
+		else if(power == 2 && prob(50))
+			triggered = TRUE
+
+	if(triggered)
+		usr.visible_message("<span class='warning'>The [src] overloads and activates!</span>")
+		activate(usr)
+
+/obj/item/device/wormhole_jaunter/proc/chasm_react(mob/user)
+	if(user.get_item_by_slot(slot_belt) == src)
+		user.visible_message("<span class='warning'>[user]'s [src] activates, saving them from the chasm!</span>")
+		activate(user)
 	else
-		user.visible_message("<span class='notice'>[user.name] activates the [src.name]!</span>")
-		var/list/L = list()
-		for(var/obj/item/device/radio/beacon/B in world)
-			var/turf/T = get_turf(B)
-			if(T.z == ZLEVEL_STATION)
-				L += B
-		if(!L.len)
-			user << "<span class='notice'>The [src.name] failed to create a wormhole.</span>"
-			return
-		var/chosen_beacon = pick(L)
-		var/obj/effect/portal/wormhole/jaunt_tunnel/J = new /obj/effect/portal/wormhole/jaunt_tunnel(get_turf(src), chosen_beacon, lifespan=100)
-		J.target = chosen_beacon
-		try_move_adjacent(J)
-		playsound(src,'sound/effects/sparks4.ogg',50,1)
-		qdel(src)
+		user.visible_message("<span class='warning'>The [src] is not attached to [user]'s belt, preventing it from saving them from the chasm. RIP.</span>")
```

Comment:
Likewise. I think it will lead to amusing "OH SHIT HE DIED" moments followed by them responding on the radio